lib/DocumentManagerInterface.php

```diff
@@ -122,4 +124,56 @@ public function createQuery($query, $locale = null, array $options = []);
      * http://doctrine-phpcr-odm.readthedocs.org/en/latest/reference/query-builder.html
      */
     public function createQueryBuilder();
+
+    /**
+     * Retrieve the document inspector.
+     *
+     * NOTE: In the future this will be a factory for retrieving inspectors
+     * for a given object/class.
```

Comment:
Ok, that's why there is this one factory class at the moment... Don't like the idea, since the `DocumentInspector` is just a temporary hack IMO. The document should be able to tell stuff like that on its own, there is also no real equivalent to the `DocumentInspector` in the ORM.